src/main/kotlin/cacophony/controlflow/CFGUtils.kt

```diff
@@ -80,12 +80,26 @@ fun not(node: CFGNode) = CFGNode.LogicalNot(node)
 
 fun writeRegister(register: Register, value: CFGNode) = CFGNode.Assignment(registerUse(register), value)
 
-class CFGFragmentBuilder {
+fun wrapAllocation(allocation: VariableAllocation): CFGNode.LValue =
+    when (allocation) {
+        is VariableAllocation.InRegister -> registerUse(allocation.register)
+        is VariableAllocation.OnStack -> memoryAccess(registerUse(rbp) sub integer(allocation.offset))
+    }
+
+class CFGFragmentBuilder() {
     private val labels: MutableMap<String, CFGLabel> = mutableMapOf()
     private val vertices = mutableMapOf<CFGLabel, CFGVertex>()
+    private val resultRegister = Register.VirtualRegister()
+    private var callConvention: CallConvention = SystemVAMD64CallConvention
```

Comment:
I left it as such so that someone can set the callConvention to test using small instances. But to be honest I don't expect anyone doing that, there should be examples covering everythink and no need to do hand-made instances.

I will delete this and leave hardcoded callConvention.